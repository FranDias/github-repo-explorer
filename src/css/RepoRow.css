.repo-row {
  display: flex;
  padding-bottom: 1em;
}

.repo-row .rotate-90 {
  transform: rotate(90deg);
}
