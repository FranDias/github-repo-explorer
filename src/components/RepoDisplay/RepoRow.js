import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import "./../../css/RepoRow.css";

class RepoRow extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false
    };
  }

  static propTypes = {
    repo: PropTypes.shape({
      name: PropTypes.string,
      created_at: PropTypes.string,
      description: PropTypes.string,
      language: PropTypes.string,
      license: PropTypes.object,
      stargazers_count: PropTypes.number,
      stargazers_url: PropTypes.string,
      forks_count: PropTypes.number,
      forks_url: PropTypes.string,
      url: PropTypes.string,
      open_issues_count: PropTypes.number,
      issues_url: PropTypes.string
    })
  };

  renderChosenAttributes() {
    const {
      url,
      name,
      description,
      stargazers_count,
      forks_count,
      open_issues_count,
      open_issues_url
    } = this.props;

    const chonesAttrs = {
      name: { lable: "name", url: url, text: name },
      description: { label: "description", text: description, expands: true },
      stargazers_count: { lable: "‚≠ê", text: stargazers_count },
      forks_count: { label: "forks", text: forks_count },
      open_issues_count: {
        label: "Open issues",
        text: open_issues_count,
        url: open_issues_url
      }
    };
  }

  renderExpandButton() {
    const { expanded } = this.state;
    const className = expanded ? "rotate-90" : "";
    return (
      <button onClick={() => this.setState({ expanded: !expanded })}>
        <span className={`expand-pointer ${className}`}>‚ñ∂</span>
      </button>
    );
  }

  render() {
    const { repo } = this.props;
    return (
      <span className="repo-row">
        <span>{repo.name}</span>,
        <span role="img" aria-labelledby="star">
          ‚≠ê
        </span>
        {repo.stargazers_count},
        <span role="img" aria-labelledby="fork">
          üç¥
        </span>
        {repo.forks_count},
        <span role="img" aria-labelledby="alarm">
          üö®
        </span>
        {repo.open_issues_count}
      </span>
    );
  }
}

export default RepoRow;
