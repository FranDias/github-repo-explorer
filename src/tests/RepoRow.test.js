import React from "react";
import renderer from "react-test-renderer";
import RepoRow from "../components/RepoDisplay/RepoRow";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const repoRowProps = {
  url: "http://frandi.as",
      name: "github-repo-explorer",
      description: "good way to play with jest",
      stargazers_count: 0, // lets be real
      forks_count: 0,
      open_issues_count: 5,
      open_issues_url: "https://github.com/frandias/github-repo-explorer/issues"
};

describe("RepoCell", () => {
  describe("snapshots", () => {
    it("matches its snapshot with no props", () => {
      const component = renderer.create(<RepoRow/>);
      expect(component).toMatchSnapshot();
    });

    it("matches snapshot with props", () => {
      const component = renderer.create(<RepoRow {...repoRowProps} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe("renders props", () => {
    Object.keys(repoRowProps).map(key => {
      it(`renders ${key}`, () => {
        const component = mount(<RepoRow {...repoRowProps} />);
        expect(component.find(RepoRow).props()[key]).toEqual(
          repoRowProps[key]
        );
      });
    });
  });
});
