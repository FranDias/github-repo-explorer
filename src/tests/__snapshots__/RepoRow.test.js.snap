// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCell snapshots matches its snapshot with no props 1`] = `null`;

exports[`RepoCell snapshots matches snapshot with props 1`] = `null`;
